### PR TITLE
feat: add letterspacing and lineheight to Typeset component

### PR DESCRIPTION
**Pull request by Ryuno-Ki**
Signed-off-by: André Jaenisch <andre.jaenisch@posteo.de>

Issue: #10560

## What I did

I've added lineHeight and letterSpacing prop to Typeset.

## How to test

- Is this testable with Jest or Chromatic screenshots?
- Does this need a new example in the kitchen sink apps?
- Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

I've extended the Typeset story accordingly.

N.B.: Would there be value in typing these properties using `CSSProperties` of React types instead of string / number? This is likely out-of-scope for this PR, though.

### CODE REVIEW

**Comment by Ryuno-Ki** [comment]
Is this something I need to take care of?

**Comment by Ryuno-Ki** [comment]
https://app.circleci.com/pipelines/github/storybookjs/storybook/19383/workflows/7ee232aa-ae45-4b13-9b26-4c0bf33c38f0/jobs/326041?invite=true#step-105-245 (E2E test) looks like not caused by this PR.

**Comment by Ryuno-Ki** [comment]
Danger JS is in trouble:

> Could not add a commit status, the GitHub token for Danger does not have access rights.

File: lib/components/src/blocks/Typeset.tsx [hunk 1]
```diff
@@ -48,6 +50,8 @@ export const Typeset: FunctionComponent<TypesetProps> = ({
   fontFamily,
   fontSizes,
   fontWeight,
+  letterSpacing,
+  lineHeight = 1.2,
```
Comment (MichaelArestad) [inline lib/components/src/blocks/Typeset.tsx:54]: Is this a default `lineHeight`?

**Comment by ndelangen** [comment]
@MichaelArestad Will we be accepting this PR, do you think.. based on the upcoming UI revision of storybook in 7.0?

**Comment by MichaelArestad** [comment]
@ndelangen I think this is okay, but the lineheight default should probably be `1.5`.  I would rather be a bit generous with the default.

**Comment by Ryuno-Ki** [comment]
Totally forgot about this PR :sweat_smile: 

Not sure, when I will have time to pick it up again.

**Comment by alicelopes47** [comment]
could we merge that please 🥺

**Pull request closed without merging**